Include API prop and improve tests

ramon-villain opened the pull request:
Hi, I'm glad you started this repo, I've been planning to do a similar thing for a while.

To summarize some of the things I did:

- Fixed some typos
- Added tests to both Provider and useFlagsmith (the coverage is 100% for those in terms of lines, but not in terms of statements because I did not have time to think of an approach to mock onChange coming from flagsmith)
- I added a prop for "API" since it's crucial if you are running locally or thru a different server, for instance, you could be on localhost:8000 or https://api-dev.flagsmith.com/api/v1 

I was also wondering:
- Do you have plans to write the module in TS? (I saw that you added the type definitions and you have PropTypes)
- Would you mind defining some linting/prettier? I personally use 4 space and semicolons, which I think it's not your standard, so I really don't want the code to look inconsistent
- Would you be up to use `jest-dom` alongside RTL?

ramon-villain committed:
Include API prop and improve tests

TheMagoo73 commented:
Hey @ramon-villain - thanks so much for jumping in, it really means a lot! And for adding the tests that so far I'd been too lazy to add, I can stop beating myself up about it now which is awesome :heart:.

- I hadn't planned on a TS re-write, but I'd be up for it as a good excuse to learn TS. The types file was added as a couple of people had requested it, and it's useful or driving Intelisense in VS Code.
- Linting and Prettier we're next on my list along side adding the tests. I should have done it earlier - but I never expected anyone to find my little repo 😆 
- Pulling in `jest-dom` sounds like a great idea!

ramon-villain committed:
Rebase + Linting + Minor test

ramon-villain commented:
@TheMagoo73 I'm glad to help, I know that react support is a pain point to use Flagsmith with React since the old days of bullet-train and I think your purpose of making it more accessible and simpler is really important. I don't know how many times I had to go into old projects and copy all those `useEffect + flagsmith.init` pieces of code 😅

I just rebased master, part of my changes are:

- Added jest-dom
- Ran the prettier/linter across all files
- Set `{ engines: { npm: >7 } }` since you are using the lockVersion 2 (good reminder for me to update npm 😂)
- Added runtime to babel in rollup

I really dislike contributing in such big chunks, so if you prefer I can open small PRs with smaller scopes instead of this big boy.

Cheers!
